library/build.gradle.kts

```diff
@@ -10,13 +9,9 @@ dependencies {
   testImplementation(libs.assertk)
 }
 
-java {
-  toolchain.languageVersion.set(JavaLanguageVersion.of(11))
-}
-
 // Used on CI to prevent publishing of non-snapshot versions.
 tasks.register("throwIfVersionIsNotSnapshot") {
-  val libraryVersion = properties["VERSION_NAME"] as String
+  val libraryVersion = providers.gradleProperty("VERSION_NAME").orNull.orEmpty()
```

Comment:
TIL

Reply:
This actually didn't work. `providers.gradleProperty("VERSION_NAME").orNull.orEmpty()` returns an empty string. Know what might be up?

Reply:
I know why, cause you defined the const in sub-project's `gradle.properties`, there's a bug of picking up it in Gradle, see https://github.com/gradle/gradle/issues/23572


https://github.com/saket/file-size/blob/80dd4f2ea411036b7ff8cdf1f9eca165863ccee5/library/gradle.properties#L7